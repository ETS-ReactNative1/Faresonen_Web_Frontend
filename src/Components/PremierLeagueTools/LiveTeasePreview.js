import React from "react";
import PropTypes from "prop-types";
import { TextArea, Grid, Header, Segment } from "semantic-ui-react";
import altOmFotballMatchService from "../../services/altOmFotballMatchService";

const LiveTeasePreview = props => {
  return (
    <Segment>
      <Header as="h3">Preview</Header>
      <Grid columns={2}>
        <Grid.Column>
          {props.selectedMatch ? <p>Kamp valgt: {props.selectedMatch}</p> : ""}
          {props.matchTimeText && props.matchTime ? (
            <p>
              Fritekst: {props.matchTimeText} {props.matchTime}
            </p>
          ) : (
            ""
          )}
          {props.channels && props.channels.length > 0 ? (
            <p>
              Kanal(er):{" "}
              {props.channels
                .map(channel =>
                  altOmFotballMatchService.getChannelName(
                    props.allChannels,
                    channel
                  )
                )
                .join(",")}
            </p>
          ) : (
            ""
          )}
          {props.homeColor && (
            <p>
              Farge hjemmelag:{" "}
              <span style={{ background: props.homeColor.hex }}>
                {props.homeColor.text}
              </span>
            </p>
          )}

          {props.awayColor && (
            <p>
              Farge bortelag:{" "}
              <span style={{ background: props.awayColor.hex }}>
                {props.awayColor.text}
              </span>
            </p>
          )}
        </Grid.Column>
        <Grid.Column>
          <TextArea
            style={{ width: "100%", color: "#298aef" }}
            autoHeight
            value={props.script}
          />
        </Grid.Column>
      </Grid>
    </Segment>
  );
};

LiveTeasePreview.propTypes = {
  selectedMatch: PropTypes.string.isRequired,
  matchTimeText: PropTypes.string.isRequired,
  matchTime: PropTypes.string.isRequired,
  channels: PropTypes.arrayOf(PropTypes.number).isRequired,
  allChannels: PropTypes.arrayOf(PropTypes.number).isRequired,
  script: PropTypes.string.isRequired,
  awayColor: PropTypes.shape({
    text: PropTypes.string,
    hex: PropTypes.string
  }),
  homeColor: PropTypes.shape({
    text: PropTypes.string,
    hex: PropTypes.string
  })
};

LiveTeasePreview.defaultProps = {
  awayColor: null,
  homeColor: null
};

export default LiveTeasePreview;
