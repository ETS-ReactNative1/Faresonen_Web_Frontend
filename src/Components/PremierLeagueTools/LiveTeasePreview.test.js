import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import LiveTeasePreview from "./LiveTeasePreview";

describe("<LiveTeasePreview />", () => {
  it("Should render properly", () => {
    let elem = renderer.create(
      <LiveTeasePreview
        selectedMatch="Brighton - Manchester United"
        matchTimeText="Avspark kl."
        matchTime="21.00"
        channels={[5]}
        script="S18 5"
        awayColor={{
          text: "RÃ¸d",
          hex: "#e90052"
        }}
        homeColor={{
          text: "Yellow",
          hex: "#eaff04"
        }}
      />
    );
    expect(elem).toMatchSnapshot();
    elem = renderer.create(
      <LiveTeasePreview
        selectedMatch="Brighton - Manchester United"
        matchTimeText="Avspark kl."
        matchTime="21.00"
        channels={[5]}
        script="S18 5"
      />
    );
    expect(elem).toMatchSnapshot();
  });
});
