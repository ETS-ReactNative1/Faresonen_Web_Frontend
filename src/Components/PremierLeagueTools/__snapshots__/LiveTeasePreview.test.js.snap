// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LiveTeasePreview /> Should render properly 1`] = `
<div
  className="ui segment"
>
  <h3
    className="ui header"
  >
    Preview
  </h3>
  <div
    className="ui two column grid"
  >
    <div
      className="column"
    >
      <p>
        Kamp valgt: 
        Brighton - Manchester United
      </p>
      <p>
        Fritekst: 
        Avspark kl.
         
        21.00
      </p>
      <p>
        Kanal(er):
         
        TV 2 Sport Premium
      </p>
      <p>
        Farge hjemmelag:
         
        <span
          style={
            Object {
              "background": "#eaff04",
            }
          }
        >
          Yellow
        </span>
      </p>
      <p>
        Farge bortelag:
         
        <span
          style={
            Object {
              "background": "#e90052",
            }
          }
        >
          RÃ¸d
        </span>
      </p>
    </div>
    <div
      className="column"
    >
      <textarea
        onChange={[Function]}
        onInput={[Function]}
        rows={3}
        style={
          Object {
            "color": "#298aef",
            "resize": "none",
            "width": "100%",
          }
        }
        value="S18 5"
      />
    </div>
  </div>
</div>
`;

exports[`<LiveTeasePreview /> Should render properly 2`] = `
<div
  className="ui segment"
>
  <h3
    className="ui header"
  >
    Preview
  </h3>
  <div
    className="ui two column grid"
  >
    <div
      className="column"
    >
      <p>
        Kamp valgt: 
        Brighton - Manchester United
      </p>
      <p>
        Fritekst: 
        Avspark kl.
         
        21.00
      </p>
      <p>
        Kanal(er):
         
        TV 2 Sport Premium
      </p>
    </div>
    <div
      className="column"
    >
      <textarea
        onChange={[Function]}
        onInput={[Function]}
        rows={3}
        style={
          Object {
            "color": "#298aef",
            "resize": "none",
            "width": "100%",
          }
        }
        value="S18 5"
      />
    </div>
  </div>
</div>
`;
